src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Asp35/IntegratedPipeline/ExecuteStepWrapper.cs

```diff
@@ -77,6 +77,13 @@ public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall ins
             if (httpContext == null)
                 throw new NullReferenceException("httpContext");
 
+            // Avoid instrumenting OPTIONS pre-flight requests
+            if ("OPTIONS".Equals(httpContext.Request?.HttpMethod, StringComparison.OrdinalIgnoreCase))
```

Comment:
Nice! I would have written it the other way around, but I like the utility of this